### PR TITLE
Reworked retro mode

### PR DESCRIPTION
**Pull request by mrange**
## Summary of the Pull Request

Reworked retro mode. This is a proposal and I expect feedback will be "mixed" and most likely rejected :).

The change is primarily in the pixel shader code but also a change of semantics of `_ComputePixelShaderSettings` in that it updates the pixel shader settings resource as this is now needed in several places.

Pixelshader code was tested manually,

 
## References

https://github.com/microsoft/terminal/issues/6191

(I think with some more tweaks we could very close to the examples in the issue)

## PR Checklist

(These are not completed yet, will do obviously if the PR seems interesting)

* [ ] Closes #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments

1. Simulates CRT "bulge"
2. Reworked scan line filter
3. Removed blur as it distracted too much in my humble opinion. I would
like to re-add a more subtle blur but was unable to get it to look right
without taking too much GPU time.
4. Simulates broken CRT by distorting colors across x/y

I also believe the new retro look should be less GPU intensive due to
the removal of the blur function (downsampling is expensive).

The bulge causes issues with copy-paste (as the coordinates don't match). Could be solved by transforming the coordinates.

## Validation Steps Performed

Manual steps:
1. Enabling retro effect `"experimental.retroTerminalEffect": true`.
2. Checking the retro effects looks as expected.
3. Checking that settings is reloaded live

### CODE REVIEW

**Comment by mrange** [comment]
image

File: src/renderer/dx/DxRenderer.cpp [hunk 1]
```diff
@@ -341,11 +341,24 @@ HRESULT DxEngine::_SetupTerminalEffects()
...
+        // Set the display resolution
+        _pixelShaderSettings.Width = 1.0f*_displaySizePixels.width<UINT>();
+        _pixelShaderSettings.Height = 1.0f*_displaySizePixels.height<UINT>();
 
-    // Gaussian distribution sigma used for blurring.
-    _pixelShaderSettings.ScaledGaussianSigma = _scale * 2.0f;
+        // Set the background
+        _pixelShaderSettings.Background = _backgroundColor;
+
+        try
```
Comment (mrange) [inline src/renderer/dx/DxRenderer.cpp:356]: I moved the resource update here as this is now needed to be updated on background change as well DPI change

File: src/renderer/dx/ScreenPixelShader.h [hunk 2]
```diff
@@ -4,84 +4,170 @@
 const char screenPixelShaderString[] = "";
 #else
 const char screenPixelShaderString[] = R"(
+// In order to run in KodeLife (great Shader IDE and free with some nagging), define the following:
```
Comment (mrange) [inline src/renderer/dx/ScreenPixelShader.h:7]: Perhaps we like to keep the old effect around as well but was uncertain on how to proceed to keep both.

File: src/renderer/dx/ScreenPixelShader.h [hunk 3]
```diff
@@ -4,84 +4,170 @@
 const char screenPixelShaderString[] = "";
 #else
 const char screenPixelShaderString[] = R"(
+// In order to run in KodeLife (great Shader IDE and free with some nagging), define the following:
+// #define KODELIFE
+
```
Comment (mrange) [inline src/renderer/dx/ScreenPixelShader.h:9]: The shader can be modified in multiple ways such as how extreme are the bulge and scan lines?

**Comment by mrange** [comment]
As the contributing guideline said it was ok to set as ready for review even though it's not complete (it's not!) I am interested in feedback if this start seems like a desired change with some tweaks.

**Comment by DHowett** [comment]
shit, this looks really good

I think we need to get our shader story in a better place before we take on too many more of these :smile:

**Comment by skyline75489** [comment]
This is more like it! The [cool-retro-term](https://github.com/Swordfish90/cool-retro-term) style in WT! 👏 👏 👏

**Comment by mrange** [comment]
@DHowett Yes I was actually considering remaking this into a generic "custom shader" feature where retro effect is just one out of possibly many.

**Comment by mrange** [comment]
Replaced by this PR: https://github.com/microsoft/terminal/pull/7058

**Pull request closed without merging**